File: leetcode/lc-us/cpp/2367.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int arithmeticTriplets(vector<int> &nums, int diff) {
        int cnt = 0;
        unordered_map<int, int> M;
        for (int x: nums) {
            cnt += M[x - diff] * M[x - diff * 2];
            M[x]++;
        }
        return cnt;
    }
};

File: leetcode/lc-us/rust/2367.rs
impl Solution {
    pub fn arithmetic_triplets(nums: Vec<i32>, diff: i32) -> i32 {
        let mut cnt: Vec<i32> = vec![0; 201];
        for x in nums {
            cnt[x as usize] += 1;
        }
        let mut res = 0;
        for i in 2..201 {
            if i - diff * 2 >= 0 && cnt[i as usize] > 0 && cnt[(i - diff) as usize] > 0 && cnt[(i - diff * 2) as usize] > 0 {
                res += cnt[(i - diff * 2) as usize] * cnt[(i - diff) as usize] * cnt[i as usize];
            }
        }
        res
    }
}